### ISSUE TITLE
clean logstash-core.gemspec dependencies version constrains

### ISSUE THREAD

**Issue by colinsurprenant** [elastic/logstash#3155]
building the logstash-core produces lots of warnings:
```sh
WARNING:  open-ended dependency on cabin (>= 0.7.0) is not recommended
  if cabin is semantically versioned, use:
    add_runtime_dependency 'cabin', '~> 0.7', '>= 0.7.0'
WARNING:  open-ended dependency on pry (>= 0) is not recommended
  if pry is semantically versioned, use:
    add_runtime_dependency 'pry', '~> 0'
WARNING:  open-ended dependency on stud (>= 0) is not recommended
  if stud is semantically versioned, use:
    add_runtime_dependency 'stud', '~> 0'
WARNING:  open-ended dependency on clamp (>= 0) is not recommended
  if clamp is semantically versioned, use:
    add_runtime_dependency 'clamp', '~> 0'
WARNING:  open-ended dependency on filesize (>= 0) is not recommended
  if filesize is semantically versioned, use:
    add_runtime_dependency 'filesize', '~> 0'
WARNING:  pessimistic dependency on treetop (~> 1.4.0) may be overly strict
  if treetop is semantically versioned, use:
    add_runtime_dependency 'treetop', '~> 1.4', '>= 1.4.0'
WARNING:  open-ended dependency on mime-types (>= 0) is not recommended
  if mime-types is semantically versioned, use:
    add_runtime_dependency 'mime-types', '~> 0'
WARNING:  open-ended dependency on rack (>= 0) is not recommended
  if rack is semantically versioned, use:
    add_runtime_dependency 'rack', '~> 0'
WARNING:  open-ended dependency on sinatra (>= 0) is not recommended
  if sinatra is semantically versioned, use:
    add_runtime_dependency 'sinatra', '~> 0'
WARNING:  open-ended dependency on minitar (>= 0) is not recommended
  if minitar is semantically versioned, use:
    add_runtime_dependency 'minitar', '~> 0'
WARNING:  open-ended dependency on jrjackson (>= 0) is not recommended
  if jrjackson is semantically versioned, use:
    add_runtime_dependency 'jrjackson', '~> 0'
WARNING:  pessimistic dependency on rspec (~> 2.14.0, development) may be overly strict
  if rspec is semantically versioned, use:
    add_development_dependency 'rspec', '~> 2.14', '>= 2.14.0'
WARNING:  open-ended dependency on logstash-devutils (>= 0, development) is not recommended
  if logstash-devutils is semantically versioned, use:
    add_development_dependency 'logstash-devutils', '~> 0'
WARNING:  See http://guides.rubygems.org/specification-reference/ for help
  Successfully built RubyGem
  Name: logstash-core
  Version: 2.0.0.dev
  File: logstash-core-2.0.0.dev-java.gem
```

**Comment by colinsurprenant** [comment]
I included the cleanup in #3111, which will close this once merged.

**Comment by colinsurprenant** [comment]
merge, closing.
